contrib/windows/msys_build.sh

```diff
@@ -31,49 +31,49 @@ checksum_download() {
 if [ -z "$ARCH" -a -z "$XC_HOST" ]; then
   export ARCH=`uname -m`
 elif [ -z "$ARCH" ]; then
-  ARCH=`echo $XC_HOST | sed 's/-w64-mingw32//'`
+  ARCH=`printf "$XC_HOST\n" | sed 's/-w64-mingw32//'`
 fi
 
-echo "" > Make.user
-echo "" > get-deps.log
+printf "\n" > Make.user
+printf "\n" > get-deps.log
 # set MARCH for consistency with how binaries get built
 if [ "$ARCH" = x86_64 ]; then
   bits=64
   archsuffix=64
   exc=seh
-  echo "override MARCH = x86-64" >> Make.user
-  echo 'USE_BLAS64 = 1' >> Make.user
-  echo 'LIBBLAS = -L$(JULIAHOME)/usr/bin -lopenblas64_' >> Make.user
-  echo 'LIBBLASNAME = libopenblas64_' >> Make.user
+  printf "override MARCH = x86-64\n" >> Make.user
+  printf "USE_BLAS64 = 1\n" >> Make.user
+  printf "LIBBLAS = -L$(JULIAHOME)/usr/bin -lopenblas64_\n" >> Make.user
+  printf "LIBBLASNAME = libopenblas64_\n" >> Make.user
 else
   bits=32
   archsuffix=86
   exc=sjlj
-  echo "override MARCH = pentium4" >> Make.user
-  echo 'LIBBLAS = -L$(JULIAHOME)/usr/bin -lopenblas' >> Make.user
-  echo 'LIBBLASNAME = libopenblas' >> Make.user
+  printf "override MARCH = pentium4\n" >> Make.user
+  printf "LIBBLAS = -L$(JULIAHOME)/usr/bin -lopenblas\n" >> Make.user
```

Comment:
the single quotes are important here, JULIAHOME is meant as the makefile variable and should not be expanded yet